docs/testing.rst

```diff
@@ -5,129 +5,131 @@ Testing Flask Applications
+format the filename like ``test_*.py``, it will be auto-discoverable by
+pytest.
+
+Next, we create a `pytest fixture`_ called
+:func:`client` that configures
+the application for testing and initializes a new database.::
 
     import os
-    from flaskr import flaskr
-    import unittest
     import tempfile
+    import pytest
+    from flaskr import flaskr
 
-    class FlaskrTestCase(unittest.TestCase):
-
-        def setUp(self):
-            self.db_fd, flaskr.app.config['DATABASE'] = tempfile.mkstemp()
-            flaskr.app.testing = True
-            self.app = flaskr.app.test_client()
-            with flaskr.app.app_context():
-                flaskr.init_db()
-
-        def tearDown(self):
-            os.close(self.db_fd)
-            os.unlink(flaskr.app.config['DATABASE'])
-
-    if __name__ == '__main__':
-        unittest.main()
-
-The code in the :meth:`~unittest.TestCase.setUp` method creates a new test
-client and initializes a new database.  This function is called before
-each individual test function is run.  To delete the database after the
-test, we close the file and remove it from the filesystem in the
-:meth:`~unittest.TestCase.tearDown` method.  Additionally during setup the
-``TESTING`` config flag is activated.  What it does is disable the error
-catching during request handling so that you get better error reports when
-performing test requests against the application.
-
-This test client will give us a simple interface to the application.  We can
-trigger test requests to the application, and the client will also keep track
-of cookies for us.
-
-Because SQLite3 is filesystem-based we can easily use the tempfile module
+    @pytest.fixture
+    def client(request):
+        db_fd, flaskr.app.config['DATABASE'] = tempfile.mkstemp()
+        flaskr.app.config['TESTING'] = True
+        client = flaskr.app.test_client()
+        with flaskr.app.app_context():
+            flaskr.init_db()
+        yield client
+        os.close(db_fd)
+        os.unlink(flaskr.app.config['DATABASE'])
+
+This client fixture will be called by each individual test.  It gives us a
+simple interface to the application, where we can trigger test requests to the
+application.  The client will also keep track of cookies for us.
+
+During setup, the ``TESTING`` config flag is activated.  What
+this does is disable error catching during request handling, so that
+you get better error reports when performing test requests against the
+application.
+
+Because SQLite3 is filesystem-based, we can easily use the :mod:`tempfile` module
 to create a temporary database and initialize it.  The
 :func:`~tempfile.mkstemp` function does two things for us: it returns a
 low-level file handle and a random file name, the latter we use as
 database name.  We just have to keep the `db_fd` around so that we can use
 the :func:`os.close` function to close the file.
 
+To delete the database after the test, we close the file and remove it
```

Comment:
When I was a n00b, I would have read this and wondered, "Does this mean I need to close the file manually somehow and then `rm` it?"

So can you make this more explicit by replacing `we close` with `the pytest fixture closes`?